telemetry/data_feature_usage.go

```diff
@@ -167,3 +176,70 @@ func GetTxnUsageInfo(ctx sessionctx.Context) *TxnUsage {
 func postReportTxnUsage() {
 	initialTxnCommitCounter = metrics.GetTxnCommitCounter()
 }
+
+// CTEUsage records the usages of CTE.
+type CTEUsage struct {
+	NonRecursiveCTEUsed uint64 `json:"nonRecursiveCTEUsed"`
+	RecursiveUsed       uint64 `json:"recursiveUsed"`
+	NonCTEUsed          uint64 `json:"nonCTEUsed"`
+}
+
+// ResetCTEUsage resets CTE usages.
+func ResetCTEUsage() error {
+	me, err := m.TelemetrySQLCTECnt.GetMetricWith(prometheus.Labels{m.LblCTEType: "nonRecurCTE"})
+	if err != nil {
```

Comment:
I think it is not a good approach to modify Prometheus metric values directly. As you can see, you have to use Gauge in order to "reset" the Prometheus metric.

It would be better to just store current values and always report a delta, and not resetting the prometheus metric value at all.